src/controller/scan/base_controller.go

```diff
@@ -699,6 +701,82 @@ func (bc *basicController) DeleteReports(ctx context.Context, digests ...string)
 	return nil
 }
 
+func (bc *basicController) GetVulnerable(ctx context.Context, artifact *ar.Artifact, allowlist allowlist.CVESet) (*Vulnerable, error) {
+	var (
+		mimeType string
+		reports  []*scan.Report
+	)
+	for _, m := range []string{v1.MimeTypeNativeReport, v1.MimeTypeGenericVulnerabilityReport} {
+		rps, err := bc.GetReport(ctx, artifact, []string{m})
+		if err != nil {
+			log.G(ctx).Errorf("get vulnerability report of the artifact %s@%s failed, error: %v", artifact.RepositoryName, artifact.Digest, err)
```

Comment:
Why do you have to print the log here as the error has been returned to the upper caller?  It seems the log can be printed in an upper hander.